addons/mrp_subcontracting/models/mrp_bom.py

```diff
@@ -7,7 +7,9 @@
 class MrpBom(models.Model):
     _inherit = 'mrp.bom'
 
-    type = fields.Selection(selection_add=[('subcontract', 'Subcontracting')])
+    type = fields.Selection(selection_add=[
+        ('subcontract', 'Subcontracting')
+    ], ondelete={'subcontract': 'default'})
```

Comment:
i would say cascade here too
ping @amoyaux

Reply:
so actually it would be weird to make them normal (subcontract bom are built by somebody else)
it would also be surprising to complete unlink them

i think the ultimate solution would be to to set them to the default and archive them.. any chance to implement that?

Reply:
I'll discuss it with @rco-odoo, maybe replacing 'cascade' by 'archive' altogether would be a generally safer option

Reply:
That being said, not every model is archivable...

Reply:
@sle-odoo After discussion with @rco-odoo, I'm gonna implement the possibility of passing a function / function name that will be called upon field unlink, this is a generic way to treat specific use-cases like yours

Reply:
@sle-odoo I propose a generic action (a callable), so you could do something like
```python
type = fields.Selection(
    selection_add=[('subcontract', 'Subcontracting')],
    ondelete={'subcontract': lambda recs: recs.write({'type': XXX, 'active': False})},
)
```

Reply:
it would completely suit our needs!